src/connectors/impls/gcl/writer.rs

```diff
@@ -0,0 +1,239 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+pub(crate) mod meta;
+mod sink;
+
+use crate::connectors::impls::gcl::writer::sink::GclSink;
+use crate::connectors::prelude::*;
+use crate::connectors::{Connector, ConnectorBuilder, ConnectorConfig, ConnectorType};
+use crate::errors::Error;
+use googapis::google::api::MonitoredResource;
+use googapis::google::logging::r#type::LogSeverity;
+use serde::Deserialize;
+use simd_json::OwnedValue;
+use std::collections::HashMap;
+use tremor_pipeline::ConfigImpl;
+
+#[derive(Deserialize, Clone)]
+pub(crate) struct Config {
+    // Optional. A default log resource name that is assigned to all log entries in entries that do not specify a value for log_name:
+    //
+    // "projects/\[PROJECT_ID]/logs/[LOG_ID\]"
+    // "organizations/\[ORGANIZATION_ID]/logs/[LOG_ID\]"
+    // "billingAccounts/\[BILLING_ACCOUNT_ID]/logs/[LOG_ID\]"
+    // "folders/\[FOLDER_ID]/logs/[LOG_ID\]"
+    // \[LOG_ID\] must be URL-encoded. For example:
+    //
+    // "projects/my-project-id/logs/syslog"
+    // "organizations/1234567890/logs/cloudresourcemanager.googleapis.com%2Factivity"
+    // The permission logging.logEntries.create is needed on each project, organization, billing account, or folder that is receiving new log entries, whether the resource is specified in logName or in an individual log entry.
+    pub log_name: Option<String>,
+
+    // Optional. A default monitored resource object that is assigned to all log entries in entries that do not specify a value for resource. Example:
+    //
+    // { "type": "gce_instance",
+    // "labels": {
+    //   "zone": "us-central1-a", "instance_id": "00000000000000000000" }}
+    #[serde(default = "default_resource")]
+    pub resource: Option<simd_json::OwnedValue>,
+
+    // Optional. Whether valid entries should be written even if some other entries fail due to INVALID_ARGUMENT or PERMISSION_DENIED errors. If any entry is not written, then the response status is the error associated with one of the failed entries and the response includes error details keyed by the entries' zero-based index in the entries.write method.
+    #[serde(default = "default_partial_success")]
+    pub partial_success: bool,
+
+    // Optional. If true, the request should expect normal response, but the entries won't be persisted nor exported. Useful for checking whether the logging API endpoints are working properly before sending valuable data.
+    #[serde(default = "default_dry_run")]
+    pub dry_run: bool,
+
+    // pub table_id: String,
+    #[serde(default = "default_connect_timeout")]
+    pub connect_timeout: u64,
+
+    // Default Log severity
+    #[serde(default = "default_log_severity")]
+    pub default_severity: Option<i32>,
+
+    #[serde(default = "default_labels")]
+    pub labels: Option<HashMap<String, String>>,
```

Comment:
It seems like using 

```suggestion
    #[serde(default = "Default::default")]
    pub labels: HashMap<String, String>,
```

instead will simplify some of the code below the `default_labels` function is no longer needed that way